### ISSUE TITLE
enum scan! fails because the following text is greedily matched

### ISSUE THREAD

**Issue by cuviper** [tinaun/serde_scan#2]
The following test is reduced from my attempt at [Advent of Code 2018 day 24](https://adventofcode.com/2018/day/24):

```rust
    #[test]
    fn scan_macro_enum() {
        #[derive(Clone, Copy, Debug, Deserialize, PartialEq)]
        #[serde(rename_all = "lowercase")]
        enum Damage {
            Fire,
            Cold,
        }

        let tests = [
            ("1 fire damage", 1, Damage::Fire),
            ("2 cold damage", 2, Damage::Cold),
        ];

        for &(test, test_n, test_damage) in &tests {
            let (n, damage): (u32, Damage) = scan!("{} {} damage" <- test).expect(test);
            assert_eq!(n, test_n);
            assert_eq!(damage, test_damage);
        }
    }
```

The "fire" test is scanned correctly, but "cold" fails.  I believe this is because the macro's `chaff` greedily matches the 'd' as if it were the start of "damage".  In the advent, "slashing" was also OK, but "bludgeoning" and "radiation" also failed with their inner 'd's.

I think the whitespace should act as a sufficient separator here, but AFAICT this is completely filtered out of both the scan string and the input.

**Comment by cuviper** [comment]
FWIW, I ended up using [`text_io::scan!`](https://docs.rs/text_io/0.1.7/text_io/macro.scan.html) instead, which is strict about matching whitespace.

**Comment by tinaun** [comment]
fixed in 0.4

time to get ready for aoc again

**Comment by cuviper** [comment]
Oh nice, thanks!
